build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -54,27 +52,40 @@ impl Goodie for GraalVM {
 
     fn is_active(&self) -> BoxFuture<'static, Result<bool>> {
         let expected_graal_version = self.graal_version.clone();
-        let expected_java_language_version = self.java_version;
         async move {
             let found_version = find_graal_version().await?;
             ensure!(found_version == expected_graal_version, "GraalVM version mismatch. Expected {expected_graal_version}, found {found_version}.");
-
-            let found_java_version = Java.check_language_version().await?;
-            ensure!(
-                found_java_version == expected_java_language_version,
-                "Java language version mismatch. Expected {expected_java_language_version}, found {found_java_version}."
-            );
-
             Result::Ok(true)
         }
         .boxed()
     }
 
     fn activation_env_changes(&self, package_path: &Path) -> Result<Vec<crate::env::Modification>> {
-        let package_path = package_path.join(self.root_directory_name());
+        let dir_entries = package_path
+            .read_dir()
+            .context("Failed to read GraalVM cache directory")
+            .unwrap()
+            .collect_vec();
+        assert_eq!(
+            dir_entries.len(),
+            1,
+            "GraalVM cache directory should contain exactly one directory"
+        );
+        let graalvm_dir = match dir_entries.get(0).unwrap() {
+            Ok(dir_entry) => dir_entry,
+            Err(err) => bail!("Failed to read GraalVM cache directory: {}", err),
+        };
+        let dir_name_tmp = graalvm_dir.file_name();
+        let dir_name = dir_name_tmp.as_str();
+        assert!(dir_name.contains("graalvm"));
+        let graal_version_str = format!(
+            "{}.{}.{}",
+            self.graal_version.major, self.graal_version.minor, self.graal_version.patch
+        );
+        assert!(dir_name.contains(&graal_version_str));
```

Comment:
In general, we never want to use `unwrap` or `assert` in non-test, as it causes a hard panic.

Instead of `unwrap` use `?`, so the function returns error.
Similarly, in place of `assert!` use `ensure!` with an appropriate message.
To make sure that there is only one element in vector (and retrieve it) you can do something along lines:
```
    let [graalvm_dir] = dir_entries.as_slice() else { bail!("GraalVM cache directory should contain exactly one directory") };
```
(also, it'd be nice to include the relevant path in the error message)

Reply:
Got it. Tried to fix it in 2a1f553404518fdfe312f3c29dd5900c7d8f43c1